### PR TITLE
Remove old PHP target

### PR DESCRIPTION
**Pull request by Simn**
I like @RealyUniqueName click the button.

CI fails because of something unrelated on development.

### CODE REVIEW

**Comment by kevinresol** [comment]
Does this mean haxe does not support php5 anymore?

**Comment by Simn** [comment]
Yes. Users who really need PHP5 can stick to Haxe 3 for a while longer. PHP5 reaches end of life in 2019, so there would only be a few months of coverage after the Haxe 4 release anyway.

**Comment by kevinresol** [comment]
Hmm.. that does sound like Haxe 4 is not going to release before mid 2018.

File: src/optimization/analyzerTexpr.ml [hunk 1]
```diff
@@ -807,7 +805,7 @@ module Fusion = struct
 							let el = List.map replace el in
 							let e2 = replace e2 in
 							e2,el
-						| Php | Cpp  when not (Common.defined com Define.Cppia) && not (Common.is_php7 com) ->
+						| Cpp ->
 							let is_php_safe e1 =
```
Comment (RealyUniqueName) [inline src/optimization/analyzerTexpr.ml]: This code should be removed also

Comment (Simn) [reply src/optimization/analyzerTexpr.ml]: Right, done!

**Comment by RealyUniqueName** [comment]
After merging this PR, `haxe -php` will stop working without `-D php7`. Should we merge it as-is? Or i can work on it moving everything (std, `#if` conditionals etc) from `php7` to `php` and merge after that.

**Comment by Simn** [comment]
Can't we just make -php define `php7` automatically?

**Comment by RealyUniqueName** [comment]
We can, but it feels like a hack. I prefer to properly move everything.

**Comment by Simn** [comment]
Fine with me.

**Comment by ncannasse** [comment]
Remember we still need to have -D php7 defined for users that use #if php7

**Comment by RealyUniqueName** [comment]
Such users have `-D php7` in their hxmls. So no additional action is required from our side.

**Comment by ncannasse** [comment]
Well if they create a new project with -php and use a library that has #if
php7 support this can't be bad. It even looks necessary.

**Comment by RealyUniqueName** [comment]
Defining `-D php7` is a requirement to use new php generator in 3.4. So they still have to add that flag to hxml of their new project. Or maybe i don't understand what you mean.

**Comment by RealyUniqueName** [comment]
Updated PR in #6612

**Pull request closed without merging**